packages/web3-providers-http/src/index.ts

```diff
@@ -0,0 +1,58 @@
+import axios, { AxiosInstance } from 'axios';
+import Web3ProviderBase from 'web3-providers-base';
+import { ProviderOptions } from 'web3-providers-base/types';
+
+import { HttpRpcOptions, HttpRpcResponse, IWeb3Provider } from '../types';
+
+export default class Web3ProvidersHttp extends Web3ProviderBase implements IWeb3Provider {
+  private _httpClient: AxiosInstance | undefined
+
+  constructor(options: ProviderOptions) {
+    super(options);
+    this._httpClient = Web3ProvidersHttp.createHttpClient(options.providerString);
+  }
+
+  static validateProviderString(providerString: string): boolean {
+    try {
+      return (typeof providerString !== 'string'
+          || /^http(s)?:\/\//i.test(providerString));
+    } catch (error) {
+      throw Error(`Failed to validate provider string: ${error.message}`);
+    }
+  }
+
+  static createHttpClient(baseUrl: string): AxiosInstance {
+    try {
+      if (!Web3ProvidersHttp.validateProviderString(baseUrl)) throw Error('Invalid HTTP(S) URI provided');
+      return axios.create({ baseURL: baseUrl });
+    } catch (error) {
+      throw Error(`Failed to create HTTP client: ${error.message}`);
+    }
+  }
+
+  setProvider(providerString: string) {
```

Comment:
This seems a bit confusing, because in the baseclass you protected the providerString, but in this class you're allowing the user to skip the bounds set on providerString. 

I need to revisit this, but I can't immediately think of a better architecture for this

Reply:
Yea, `super.setProvider(providerString);` should come first, I was just battling an architecture issue where the protected provider error was being thrown. I'm going to try rethinking this with getters/setters

Reply:
ahh! Maybe when you adjust the get/set in the other pr this one will solve itself

Reply:
Not relevant anymore - `protectProvider` was removed

Reply:
Thinking about the user flow
```
const provider = new Web3ProvidersHttp({providerString: "http://localhost"})
provider.setProvider("http://123.123.123") // actually creates new http instance
provider.providerString("http://444.444.444") // doesn't do anything
```
What benefit do we have having the provider string in the base class?